url74c52881d6_AndrewBourgeois_JGWCAConstants_tgz-pJ8-be_demmel_fun_jgwcaconstants_Tutorial4J8/src/be/demmel/fun/jgwcaconstants/bll/GWCAOperations.java

```diff
@@ -0,0 +1,2819 @@
+    public void offerItem(int itemId) throws IOException {
+        sendPacket(GWCAOperation.OFFER_ITEM, itemId);
+    }
+
+    /**
+     * Like pressing the "Cancel" button in a trade.
+     * @throws IOException 
+     */
+    public void cancelTrade() throws IOException {
+        sendPacket(GWCAOperation.CANCEL_TRADE);
+    }
+
+    /**
+     * Like pressing the "Accept" button in a trade. Can only be used after both players have submitted their offer.
+     * @throws IOException 
+     */
+    public void acceptTrade() throws IOException {
+        sendPacket(GWCAOperation.ACCEPT_TRADE);
+    }
+
+    /**
+     * attacks the specified target with weapon.
+     * @param targetId
+     * @throws IOException 
+     */
+    public void attack(int targetId) throws IOException {
+        sendPacket(GWCAOperation.ATTACK, targetId);
+    }
+
+    /**
+     * @param x
+     * @param y
+     * @throws IOException 
+     */
+    public void move(float x, float y) throws IOException {
+        LOG.debug("Moving to: {} {}", x, y);
+        sendPacket(GWCAOperation.MOVE, x, y);
+    }
+
+    /**
+     * @param x
+     * @param y
+     * @param random
+     * @throws IOException 
+     */
+    public void moveEx(float x, float y, int random) throws IOException {
+        x += -random + new Random().nextInt(random * 2);
+        y += -random + new Random().nextInt(random * 2);
+        this.move(x, y);
+    }
+
+    /**
+     * Works on all skills including attack skills. Remember that some skills that target yourself may require that specified
+     * @param skillSlot
+     * @param targetId or -1 for the current target, -2 for yourself
+     * @throws IOException 
+     */
+    public void useSkill(SkillBarSkillSlot skillSlot, int targetId) throws IOException {
+        sendPacket(GWCAOperation.USE_SKILL, skillSlot.getValue(), targetId);
+    }
+
+    /**
+     * @param weaponSlot
+     * @throws IOException 
+     */
+    public void changeWeaponSet(int weaponSlot) throws IOException {
+        sendPacket(GWCAOperation.CHANGE_WEAPON_SET, weaponSlot);
+    }
+
+    /**
+     * zones to the outpost/city specified and if district number is provided also attempts to enter that district. Always uses current region and language
+     * @param zoneId
+     * @param districtNumber
+     * @throws IOException 
+     */
+    public void zoneMap(int zoneId, int districtNumber) throws IOException {
+        sendPacket(GWCAOperation.ZONE_MAP, zoneId, districtNumber);
+    }
+
+    /**
+     * @param zoneId
+     * @param districtNumber
+     * @throws IOException 
+     */
+    public void travelTo(int zoneId, int districtNumber) throws IOException {
+        long start = System.currentTimeMillis();
+        this.initMapLoad();
+        this.zoneMap(zoneId, districtNumber);
+
+        int status = -1;
+        boolean secondStatusCheck = false;
+
+        do {
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)